test/mock_repository.dart

```diff
@@ -0,0 +1,133 @@
+  CharacterViewData(id: 7, name: "7", thumbnailUrl: "url 7", description: "description 7"),
+  CharacterViewData(id: 8, name: "8", thumbnailUrl: "url 8", description: "description 8"),
+  CharacterViewData(id: 9, name: "9", thumbnailUrl: "url 9", description: "description 9"),
+  CharacterViewData(id: 10, name: "10", thumbnailUrl: "url 10", description: "description 10"),
+  CharacterViewData(id: 11, name: "11", thumbnailUrl: "url 11", description: "description 11"),
+  CharacterViewData(id: 12, name: "12", thumbnailUrl: "url 12", description: "description 12"),
+  CharacterViewData(id: 13, name: "13", thumbnailUrl: "url 13", description: "description 13"),
+  CharacterViewData(id: 14, name: "14", thumbnailUrl: "url 14", description: "description 14"),
+  CharacterViewData(id: 15, name: "15", thumbnailUrl: "url 15", description: "description 15"),
+  CharacterViewData(id: 16, name: "16", thumbnailUrl: "url 16", description: "description 16"),
+  CharacterViewData(id: 17, name: "17", thumbnailUrl: "url 17", description: "description 17"),
+  CharacterViewData(id: 18, name: "18", thumbnailUrl: "url 18", description: "description 18"),
+  CharacterViewData(id: 19, name: "19", thumbnailUrl: "url 19", description: "description 19"),
+  CharacterViewData(id: 20, name: "20", thumbnailUrl: "url 20", description: "description 20"),
+  CharacterViewData(id: 21, name: "21", thumbnailUrl: "url 21", description: "description 21"),
+  CharacterViewData(id: 22, name: "22", thumbnailUrl: "url 22", description: "description 22"),
+  CharacterViewData(id: 23, name: "23", thumbnailUrl: "url 23", description: "description 23"),
+  CharacterViewData(id: 24, name: "24", thumbnailUrl: "url 24", description: "description 24"),
+  CharacterViewData(id: 25, name: "25", thumbnailUrl: "url 25", description: "description 25"),
+  CharacterViewData(id: 26, name: "26", thumbnailUrl: "url 26", description: "description 26"),
+  CharacterViewData(id: 27, name: "27", thumbnailUrl: "url 27", description: "description 27"),
+  CharacterViewData(id: 28, name: "28", thumbnailUrl: "url 28", description: "description 28"),
+  CharacterViewData(id: 29, name: "29", thumbnailUrl: "url 29", description: "description 29"),
+];
+
+const mockSeries = <Series>[
+  Series(id: 0, title: "title 0", thumbnailUrl: "url 0"),
+  Series(id: 1, title: "title 1", thumbnailUrl: "url 1"),
+  Series(id: 2, title: "title 2", thumbnailUrl: "url 2"),
+  Series(id: 3, title: "title 3", thumbnailUrl: "url 3"),
+  Series(id: 4, title: "title 4", thumbnailUrl: "url 4"),
+];
+
+const mockSeriesViewData = <SeriesViewData>[
+  SeriesViewData(title: "title 0", thumbnailUrl: "url 0"),
+  SeriesViewData(title: "title 1", thumbnailUrl: "url 1"),
+  SeriesViewData(title: "title 2", thumbnailUrl: "url 2"),
+  SeriesViewData(title: "title 3", thumbnailUrl: "url 3"),
+  SeriesViewData(title: "title 4", thumbnailUrl: "url 4"),
+];
+
+class MockRepository extends MarvellRepository {
+  bool dataRetrieveException = false;
```

Comment:
I suggest make this fields private and add method to set exceptions to throw:
`void mockRetreivingAnyDataWithException(Exception exception)`